### ISSUE TITLE
State explicitly in the manual that instance method static variables are private and cannot be accessed

### ISSUE THREAD

**Issue by patchuby** [YoYoGames/GameMaker-Bugs#4687]
### Description

It's not obvious from the manual that we can't access the static variables from an instance method but only from constructor and script functions. 

It could be specified in the following articles : 

- http://127.0.0.1:51290/index.htm#t=GameMaker_Language%2FGML_Reference%2FVariable_Functions%2Fvariable_struct_get.htm&rhsearch=static_get
- https://manual.gamemaker.io/monthly/en/index.htm#t=GameMaker_Language%2FGML_Overview%2FFunctions%2FStatic_Variables.htm

### Manual Link

https://manual.gamemaker.io/monthly/en/index.htm#t=GameMaker_Language%2FGML_Overview%2FFunctions%2FStatic_Variables.htm

**Comment by rwkay** [comment]
@gurpreetsinghmatharoo see my answer on the linked bug #4686 if you need any more information then let me know....

**Comment by patchuby** [comment]
Hi @YYBartT, the link to the static struct manual page doesn't work for me (and it's a bit hard to read the html version in the commit)

**Comment by YYBartT** [comment]
Ah yes, you're right, the link was missing a `http://`. That's fixed now.

**Comment by patchuby** [comment]
@YYBartT Yes the manual seems to be much more complete and explicit. To be honest it's been a month so I don't fully remember what exactly I was confused about. I do remember having a problem accessing the static struct for method defined in an instance event as I thought I could access it like "script" functions and constructors. Even though it can be logically deduced from the page why you can't access it the same way, maybe you can add a warning section to spell it out, in case people like me who read quickly through a page looking for a specific info (though it's a bad habit) can find the reason why ? That's my 2 cents anyway.

**Comment by YYBartT** [comment]
@patchuby Currently you can get the static struct of a method, though you have to pass the index of its function (which you can get with `method_get_index`). Like this: 
```gml
/// Create Event
my_method = method(self, my_func);

var _s;
_s = static_get(my_method);                    // This doesn't work (yet)
_s = static_get(method_get_index(my_method));  // This does work
```
Simply passing the method doesn't work (and will throw a "variable not found" error), though this will be added as explained in the ticket linked above.
I added a note to the manual page of `static_get` to explain that you can get the static struct through the method index.
Does this answer the question you had?
